libs/platform/ebpf_pinning_table.c

```diff
@@ -290,22 +294,22 @@ ebpf_pinning_table_enumerate_entries(
 }
 
 ebpf_result_t
-ebpf_pinning_table_get_next_name(
+ebpf_pinning_table_get_next_path(
     _In_ ebpf_pinning_table_t* pinning_table,
     ebpf_object_type_t object_type,
-    _In_z_ const char* start_name,
-    _Out_writes_z_(EBPF_MAX_PIN_PATH_LENGTH) char* next_name)
+    _In_z_ const char* start_path,
+    _Out_writes_z_(EBPF_MAX_PIN_PATH_LENGTH) char* next_path)
 {
-    if ((pinning_table == NULL) || (start_name == NULL) || (next_name == NULL)) {
+    if ((pinning_table == NULL) || (start_path == NULL) || (next_path == NULL)) {
         return EBPF_INVALID_ARGUMENT;
     }
 
-    // Copy the name provided into a buffer that we can guarantee is null terminated.
-    char name_buffer[EBPF_MAX_PIN_PATH_LENGTH];
-    strncpy_s(name_buffer, EBPF_MAX_PIN_PATH_LENGTH, start_name, _TRUNCATE);
-    const ebpf_utf8_string_t name_string = {(uint8_t*)name_buffer, strlen(name_buffer)};
-    const ebpf_utf8_string_t* name = &name_string;
-    const uint8_t* previous_key = (name->length == 0) ? NULL : (const uint8_t*)&name;
+    // Copy the path provided into a buffer that we can guarantee is null terminated.
```

Comment:
If we use ebpf_utf8_string_t we don't need to worry about null termination.

Reply:
your comments are actually about code that was already there prior to this PR.  They're fair comments, but better for a different issue I think since not related to "show programs" functionality per se.

Reply:
Fair enough, file #551 to track it.